lib/core/commands/user-commands.js

```diff
@@ -7,31 +7,51 @@ class UserCommands {
   async createUser(user) {
     const command = {
       createUser: user.user,
-      // mongo internally uses pwd
-      pwd: user.password,
       customData: user.customData || {},
       roles: user.roles || []
     }
+
+    // mongo internally uses pwd
+    if (user.password) {
+      command.pwd = user.password
+    }
+    // Use external db iff external is set
+    if (user.external) {
+      this.client.useDatabase('$external')
```

Comment:
This needs to either move up or restore the old database used, as the client will keep state

Reply:
Er even have each command define the used dB?

Reply:
I  was waiting for this, I think we need to have sync up about how we want to maintain these types of state. I know the replication commands just auth as `/admin` and then run all commands but I feel like we need to figure out how we handle which DB  commands are run against.

For example even with regular users you can add them to the `admin` db or the actual db, both work but affect how you connect to the db's using username/password auth.


See `authSource` https://mongodb.github.io/node-mongodb-native/3.6/reference/connecting/connection-settings/ 

I am not sure what to do here.